### DISCUSSION TITLE
Add support for NextJS

### DISCUSSION THREAD

**Discussion by john--kane** [gorules/zen#75] [Q&A]
Is it possible to add support for NextJS? 

Module parse failed: Unexpected character '�' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)

Import trace for requested module:
./node_modules/.pnpm/@gorules+zen-engine-darwin-x64@0.6.0/node_modules/@gorules/zen-engine-darwin-x64/zen-engine.darwin-x64.node
./node_modules/.pnpm/@gorules+zen-engine@0.6.0/node_modules/@gorules/zen-engine/index.js

**Comment by stefan-gorules** [comment]
Hi @john--kane,

This is not something that we have control over. You are able to add `node-loader` to webpack bundler in order to load native bindings. I'm not sure how this is handled in NextJS, but I would assume it's similar to adding a "pg" module (for postgres).

You need to set-up webpack externals and make sure `@gorules/zen-engine` is installed in node_modules on the machine when you run it after building.

Try to find some examples with TypeORM and Postgres ("pg") or even this looks like a decent starting point: https://github.com/vercel/next.js/issues/2790.

**Comment by justin0108** [comment]
Hi @john--kane 
I happen to run into the same issue today with integrating RuleEngine into NextJS (14.1.4).
I manage to get it working by

```
# add loader dependency
yarn add nextjs-node-loader --dev

# update next.config.mjs
/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: (config, { dev, isServer, webpack, nextRuntime }) => {
    config.module.rules.push({
      test: /\.node$/,
      use: [
        {
          loader: "nextjs-node-loader",
        },
      ],
    });
    return config;
  },
};

export default nextConfig;
```
